src/components/support-form/helpers/validation-schema.ts

```diff
@@ -1,99 +1,135 @@
+    email: yup.string().email().required(),
+    name: yup.string().required(),
+    phone: yup.string().required(),
+    address: yup.string().required(),
+  })
+  .defined()
+
+const roles: yup.SchemaOf<Roles> = yup
+  .object()
+  .shape({
+    benefactor: yup.bool(),
+    partner: yup.bool(),
+    volunteer: yup.bool(),
+    associationMember: yup.bool(),
+    promoter: yup.bool(),
+  })
+  .defined()
+
+const benefactor: yup.SchemaOf<Benefactor> = yup
+  .object()
+  .shape({
+    campaignBenefactor: yup.bool(),
+    platformBenefactor: yup.bool(),
+  })
+  .defined()
+
+const associationMember: yup.SchemaOf<Member> = yup
+  .object()
+  .shape({ isMember: yup.bool() })
+  .defined()
+
+const partner: yup.SchemaOf<Partner> = yup
+  .object()
+  .shape({
+    npo: yup.bool(),
+    bussiness: yup.bool(),
+    other: yup.bool(),
+    otherText: yup.string().when('partner.other', {
+      is: true,
+      then: yup.string(),
+    }),
+  })
+  .defined()
+
+const volunteer: yup.SchemaOf<Volunteer> = yup
+  .object()
+  .shape({
+    areas: yup.array().of(yup.string()),
+  })
+  .defined()
+
+const promoter: yup.SchemaOf<Promoter> = yup
+  .object()
+  .shape({
+    mediaPartner: yup.bool(),
+    ambassador: yup.bool(),
+    other: yup.bool(),
+    otherText: yup.string(),
+  })
+  .defined()
+
 export const validationSchema: yup.SchemaOf<SupportFormData> = yup.object().shape({
   terms: yup.bool().required().oneOf([true], 'common:support-form.termsHelperText'),
   newsletter: yup.bool().required(),
-  info: yup
-    .object()
-    .shape({
-      email: yup.string().email().required(),
-      name: yup.string().required(),
-      phone: yup.string().required(),
-      address: yup.string().required(),
-    })
-    .required(),
-  roles: yup
-    .object()
-    .shape({
-      benefactor: yup.bool(),
-      partner: yup.bool(),
-      volunteer: yup.bool(),
-      associationMember: yup.bool(),
-      promoter: yup.bool(),
-    })
-    .required()
-    .test('checkboxChecked', 'Must have at least one checked box', checkboxChecked),
+  info: info.required(),
+  roles: roles.required().test('checkboxChecked', checkboxError, checkboxChecked),
   benefactor: yup.object().when('roles.benefactor', {
     is: true,
-    then: yup
-      .object()
-      .shape({
-        campaignBenefactor: yup.bool(),
-        platformBenefactor: yup.bool(),
-      })
-      .test('checkboxChecked', 'Must have at least one checked box', checkboxChecked),
+    then: benefactor.required().test('checkboxChecked', checkboxError, checkboxChecked),
+    // otherwise: benefactor,
   }),
   associationMember: yup.object().when('roles.associationMember', {
     is: true,
-    then: yup
-      .object()
-      .shape({
-        isMember: yup.bool(),
-      })
-      .test('checkboxChecked', 'Must have at least one checked box', checkboxChecked),
+    then: associationMember.required().test('checkboxChecked', checkboxError, checkboxChecked),
+    // otherwise: associationMember,
```

Comment:
```suggestion
    then: associationMember.required().test('checkboxChecked', checkboxError, checkboxChecked),
    otherwise: associationMember.notRequired().default(undefined),
```

Reply:
That not solve type error in the validation schema, but we can add 'otherwise' as this example